portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,29 +23,54 @@
 import java.util.ArrayList;
 import com.google.gson.Gson;
 import com.google.sps.servlets.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  ArrayList<Comment> commentHistory = new ArrayList<Comment>(); 
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
+  
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery storedComments = datastore.prepare(query);
+
+    ArrayList<Comment> commentHistory = new ArrayList<Comment>(); 
+    for(Entity comment : storedComments.asIterable()){
+        commentHistory.add(new Comment(
+          (String) comment.getProperty("username"),
```

Comment:
Though experiment: What will happen to your code if you run into little Bobby Drop Tables? https://xkcd.com/327/